src/vivarium_census_prl_synth_pop/results_processing/names.py

```diff
@@ -1,47 +1,131 @@
-from typing import Dict
+from typing import Any, Dict
 
+import numpy as np
 import pandas as pd
 from vivarium import Artifact
+from vivarium.framework.randomness import RandomnessStream
 
+from vivarium_census_prl_synth_pop.constants import data_keys
 from vivarium_census_prl_synth_pop.results_processing.formatter import (
     format_data_for_mapping,
 )
+from vivarium_census_prl_synth_pop.utilities import vectorized_choice
+
+
+def get_first_name_map(
+    raw_data: Dict[str, pd.DataFrame],
+    artifact: Artifact,
+    randomness: RandomnessStream,
+) -> Dict[str, pd.Series]:
+    first_name_data = get_data_for_first_middle_name_mapping(
+        "first_name_id",
+        raw_data,
+    )
+
+    first_name_map = generate_first_and_middle_names(
+        first_name_data, "first_name", artifact, randomness
+    )
+
+    return {"first_name": first_name_map}
+
+
+def get_middle_name_map(
+    raw_data: Dict[str, pd.DataFrame], artifact: Artifact, randomness: RandomnessStream
+) -> Dict[str, pd.Series]:
+    middle_name_data = get_data_for_first_middle_name_mapping(
+        "middle_name_id",
+        raw_data,
+    )
+
+    middle_name_map = generate_first_and_middle_names(
+        middle_name_data, "middle_name", artifact, randomness
+    )
+
+    return {"middle_name": middle_name_map}
 
 
 def get_data_for_first_middle_name_mapping(
-    index_name: str, raw_data: Dict[str, pd.DataFrame]
+    index_name: str, obs_data: Dict[str, pd.DataFrame]
 ) -> pd.DataFrame:
 
     input_cols = [index_name, "date_of_birth", "sex", "random_seed"]
     output_cols = [index_name, "year_of_birth", "sex"]
 
     name_data = format_data_for_mapping(
         index_name=index_name,
-        raw_results=raw_data,
-        columns_required=input_cols,
+        obs_results=obs_data,
         output_columns=output_cols,
     )
 
     return name_data
 
 
-def get_first_name_map(
-    raw_data: Dict[str, pd.DataFrame], artifact: Artifact
-) -> Dict[str, pd.DataFrame]:
-    first_name_data = get_data_for_first_middle_name_mapping(
-        "first_name_id",
-        raw_data,
-    )
+def generate_first_and_middle_names(
+    df_in: pd.DataFrame, additional_key: Any, artifact: Artifact, randomness: RandomnessStream
+) -> pd.Series:
+    """Generate synthetic names for individuals
+    Parameters
+    ----------
+    df_in : pd.DataFrame, with columns sex, year_of_birth
+    additional_key: key to pass to randomness stream
+    artifact: Artifact with synthetic data
+    randomness: Randomness stream for results processing
+    Returns
+    -------
+    returns pd.Series with name_ids as index and string names as values.
+    """
```

Comment:
The formatting of this docstring is off. Needs a period after the summary, blank line after. You also need a blank line after the Parameters section and a blank line between the last `"""` and code. [See PEP on doc string conventions](https://peps.python.org/pep-0257/)